Control/lib/middleware/detectorOwnership.middleware.js

```diff
@@ -0,0 +1,33 @@
+const {User} = require('../dtos/User');
+/**
+   * Middleware function to check detector ownership.
+   * Based on the session object, it checks if the user has ownership of the detector lock.
+   * @param {Request} req - Express Request object.
+   * @param {Response} res - Express Response object.
+   * @param {Function} next - Next middleware to call.
+   */
+const detectorOwnershipMiddleware = (req, res, next) => {
+  const { detectorId } = req.params;
+  const { name, username, personid, access } = req.session || {};
+
+  if (!detectorId || !access) {
+    return res.status(400).json({ message: 'Invalid request: missing information' });
+  }
+  
+  try {
+    const u = new User(username, name, personid, access);
+    console.log(`Checking locks for detector ${detectorId}`);
```

Comment:
Hi, @pepijndik . We do not make use of `console` in this project but instead we use the `Logger` defined in the WebUI framework.
Nevertheless, in this case there is no need to log anything here in case of success